### PR TITLE
Fix flaky traffic_shift test - Ignore same aspath routes during validation

### PR DESCRIPTION
**Pull request by tjchadaga**
### Description of PR

Summary:
When validating routes on neighbor hosts after TSB, the routes may be different from the original list on some neighbors, due to same aspath filtering. Enhanced test to handle this scenario.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
After TSB, while comparing current routes to original routes on a neighbor, ignore the routes advertised by that neighbor.

#### How did you do it?
Ignore routes with aspath list containing the neighbor's aspath during route validation

#### How did you verify/test it?
Run bgp/test_traffic_shift.py on T1 multiple times

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/bgp/test_traffic_shift.py [hunk 1]
```diff
@@ -231,25 +231,47 @@ def verify_current_routes_announced_to_neighs(dut_host, neigh_hosts, orig_routes
...
+def check_and_log_routes_diff(duthost, neigh_hosts, orig_routes_on_all_nbrs, cur_routes_on_all_nbrs,ip_ver):
     cur_nbrs = set(cur_routes_on_all_nbrs.keys())
     orig_nbrs = set(orig_routes_on_all_nbrs.keys())
     if cur_nbrs != orig_nbrs:
         logger.warn("Neighbor list mismatch: {}".format(cur_nbrs ^ orig_nbrs))
         return False
-
+    
+    routes_dut = parse_rib(duthost, ip_ver)
     for hostname in orig_routes_on_all_nbrs.keys():
```
Comment (StormLiangMS) [inline tests/bgp/test_traffic_shift.py:246]: I didn't see why we need embedded for loop here @tjchadaga

File: tests/bgp/test_traffic_shift.py [hunk 2]
```diff
@@ -231,25 +231,47 @@ def verify_current_routes_announced_to_neighs(dut_host, neigh_hosts, orig_routes
...
-                            set(cur_routes_on_all_nbrs[hostname]) - set(orig_routes_on_all_nbrs[hostname])))
+        if orig_routes_on_all_nbrs[hostname] != cur_routes_on_all_nbrs[hostname]:            
+            routes_diff =set(orig_routes_on_all_nbrs[hostname]) ^ set(cur_routes_on_all_nbrs[hostname])                
+        all_diffs_in_host_aspath = True
+        for hostname in orig_routes_on_all_nbrs.keys():
+            if orig_routes_on_all_nbrs[hostname] != cur_routes_on_all_nbrs[hostname]:
+                routes_diff =set(orig_routes_on_all_nbrs[hostname]) ^ set(cur_routes_on_all_nbrs[hostname])
+                for route in routes_diff:
+                    if route not in routes_dut.keys():
+                            all_diffs_in_host_aspath = False
```
Comment (StormLiangMS) [inline tests/bgp/test_traffic_shift.py:255]: to use 4 space as intents @tjchadaga

File: tests/bgp/test_traffic_shift.py (hunk 1 above)
Comment (tjchadaga) [reply tests/bgp/test_traffic_shift.py:247]: this was added by mistake during cleanup. Thanks for flagging!

File: tests/bgp/test_traffic_shift.py (hunk 2 above)
Comment (tjchadaga) [reply tests/bgp/test_traffic_shift.py]: fixed

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/bgp/test_traffic_shift.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by yejianquan** [comment]
@tjchadaga Can you please resolve the conflict so that we can merge it into the 202205 branch?

**Pull request merged**